jkube-kit/build/service/buildpacks/src/test/java/org/eclipse/jkube/kit/service/buildpacks/AbstractBuildPackCliDownloaderTest.java

```diff
@@ -0,0 +1,240 @@
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+abstract class AbstractBuildPackCliDownloaderTest {
+  private static final String TEST_PACK_VERSION = "0.32.1";
+  private KitLogger kitLogger;
+  @TempDir
+  private File temporaryFolder;
+  private File remoteDirectory;
+  private File oldPackCliInJKubeDir;
+  private TestHttpStaticServer server;
+  private BuildPackCliDownloader buildPackCliDownloader;
+
+  abstract String getApplicablePackBinary();
+  abstract String getInvalidApplicablePackBinary();
+  abstract String getPlatform();
+  abstract String getProcessorArchitecture();
+
+  @BeforeEach
+  void setUp() {
+    kitLogger = spy(new KitLogger.SilentLogger());
+    Map<String, String> overriddenSystemProperties = new HashMap<>();
+    Map<String, String> overriddenEnvironmentVariables = new HashMap<>();
+    overriddenSystemProperties.put("user.home", temporaryFolder.getAbsolutePath());
+    overriddenSystemProperties.put("os.name", getPlatform());
+    overriddenSystemProperties.put("os.arch", getProcessorArchitecture());
+    overriddenEnvironmentVariables.put("HOME", temporaryFolder.getAbsolutePath());
+    overriddenEnvironmentVariables.put("PATH", temporaryFolder.toPath().resolve("bin").toFile().getAbsolutePath());
+    EnvUtil.overridePropertyGetter(overriddenSystemProperties::get);
+    EnvUtil.overrideEnvGetter(overriddenEnvironmentVariables::get);
+    remoteDirectory = new File(Objects.requireNonNull(getClass().getResource("/artifacts")).getFile());
+    server = new TestHttpStaticServer(remoteDirectory);
+    buildPackCliDownloader = new BuildPackCliDownloader(kitLogger, createPackProperties(server));
+  }
+
+  @AfterEach
+  void tearDown() throws IOException {
+    server.close();
+    EnvUtil.overridePropertyGetter(System::getProperty);
+    EnvUtil.overrideEnvGetter(System::getenv);
+  }
+
+  @Test
+  @DisplayName("pack binary exists and is valid, then use already downloaded cli")
+  void givenOldPackCliExistsAndIsValid_thenAlreadyExistingPackCliReturned() throws IOException {
+    // Given
+    oldPackCliInJKubeDir = new File(Objects.requireNonNull(getClass().getResource(String.format("/%s", getApplicablePackBinary()))).getFile());
+    givenPackCliAlreadyDownloaded();
+
+    // When
+    File downloadedCli = buildPackCliDownloader.getPackCLIIfPresentOrDownload();
+
+    // Then
+    assertThat(downloadedCli.toPath()).isEqualTo(temporaryFolder.toPath().resolve(".jkube").resolve(getApplicablePackBinary()));
+    assertThat(downloadedCli).hasSameTextualContentAs(oldPackCliInJKubeDir);
+  }
+
+  @SuppressWarnings("unused")
+  @Nested
+  @DisplayName("pack binary doesn't exist")
+  class PackBinaryDoesNotExist {
+    @Nested
+    @DisplayName("download initiated due to no existing pack binary")
+    class DownloadDueToNoExistingPackBinary extends PackBinaryDownload { }
```

Comment:
nit: it's very unclear what's actually happening here. I understand that when the tests are run, something similar to the structure that was suggested in https://github.com/eclipse/jkube/pull/2495#discussion_r1433833972 will be showed

Reply:
It's shown like this
![Screenshot_20240117_192808](https://github.com/eclipse/jkube/assets/13834498/741f67cf-2dfa-4ae7-95dc-acbf3554f7b3)